### PR TITLE
Add custom runners

### PR DESCRIPTION
**Pull request by kgugala**
This PR enables custom GH-Actions runners for this repository. It also adds nexys_video test (which was breaking GH hosted runner)

### CODE REVIEW

File: .github/scripts/common.sh [hunk 1]
```diff
@@ -17,15 +17,5 @@
 # SPDX-License-Identifier: Apache-2.0
 
 function tuttest_exec() {
```
Comment (mithro) [inline .github/scripts/common.sh:19]: Why is this removed?

Comment (pgielda) [reply .github/scripts/common.sh:19]: Because it makes the logs completely unreadable

Comment (pgielda) [reply .github/scripts/common.sh:19]: Adding WIP tag btw, we are still experimenting here

Comment (pgielda) [reply .github/scripts/common.sh:19]: This is back in a more concise form.

**Comment by pgielda** [comment]
Can we rebase and get it merged @kgugala  ?

**Comment by kgugala** [comment]
> Can we rebase and get it merged @kgugala ?

rebased, will merge it when CI goes green
